tianshou/policy/base.py

```diff
@@ -216,9 +220,23 @@ def compute_nstep_return(
         batch.returns = target_q * gammas + returns
         # prio buffer update
         if isinstance(buffer, PrioritizedReplayBuffer):
-            batch.update_weight = buffer.update_weight
-            batch.indice = indice
             batch.weight = to_torch_as(batch.weight, target_q)
         else:
             batch.weight = torch.ones_like(target_q)
         return batch
+
+    def post_process_fn(self, batch: Batch,
+                        buffer: ReplayBuffer, indice: np.ndarray):
+        """Post-process the data from the provided replay buffer. Typical
+        usage is to update the sampling weight in prioritized experience
+        replay. Check out :ref:`policy_concept` for more information.
+        """
+        if isinstance(buffer, PrioritizedReplayBuffer):
+            buffer.update_weight(indice, batch.weight)
+
+    def update(self, sample_size: int, buffer: ReplayBuffer, *args, **kwargs):
```

Comment:
```suggestion
    def update(self, batch_size: int, buffer: ReplayBuffer, *args, **kwargs):
        """Update the policy network and replay buffer (if needed). It includes
        three function steps: process_fn, learn, and post_process_fn.

        :param int batch_size: 0 means it will extract all the data from the
            buffer, otherwise it will sample a batch with the given batch_size.
        :param ReplayBuffer buffer: the corresponding replay buffer.
        """
```